aten/src/ATen/native/mkldnn/Normalization.cpp

```diff
@@ -0,0 +1,72 @@
+#include <ATen/ATen.h>
+#include <ATen/Config.h>
+#include <ATen/NativeFunctions.h>
+#include <tuple>
+
+#if !AT_MKLDNN_ENABLED()
+
+namespace at {
+namespace native {
+
+std::tuple<Tensor, Tensor, Tensor> mkldnn_batch_norm(
+    const Tensor& self,
+    const Tensor& weight,
+    const Tensor& bias,
+    const Tensor& running_mean,
+    const Tensor& running_var,
+    bool train,
+    double momentum,
+    double eps) {
+  AT_ERROR("mkldnn_batch_norm: ATen not compiled with MKLDNN support");
+}
+
+} // namespace native
+} // namespace at
+
+#else // AT_MKLDNN_EBABLED
+
+#include <ATen/native/mkldnn/MKLDNNCommon.h>
+
+namespace at {
+namespace native {
+
+std::tuple<Tensor, Tensor, Tensor> mkldnn_batch_norm(
+    const Tensor& input,
+    const Tensor& weight,
+    const Tensor& bias,
+    const Tensor& running_mean,
+    const Tensor& running_var,
+    bool train,
+    double momentum,
+    double eps) {
+  ideep::tensor& x = itensor_from_mkldnn(input);
+  ideep::tensor& w = itensor_from_mkldnn(weight);
+  ideep::tensor& b = itensor_from_mkldnn(bias);
+  ideep::tensor& m = itensor_from_mkldnn(running_mean);
+  ideep::tensor& v = itensor_from_mkldnn(running_var);
+
+  ideep::tensor y;
+
+  if (train) {
```

Comment:
This training support is incomplete. Derivatives.yaml calls native_batch_norm_backward and that one doesn't know what to do with mkldnn tensors. Thus I'd say either assert that train=false or implement native_batch_norm_backward for consistency

Reply:
Makes senses will add an assert